### PR TITLE
Add mt-tenant e2e test

### PR DESCRIPTION
**Pull request by steven0711dong**
Fixes #

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by matzew** [comment]
@lionelvillard @steven0711dong  what about this PR?

since some of the tests are now REKT test... any thoughts about closing this?

**Comment by steven0711dong** [comment]
@matzew this PR is on top of the recently merged PR. This PR adds one additional test case. Lionel and I are working on merging this PR.

File: test/rekt/features/kafkasource/data_plane.go [hunk 1]
```diff
@@ -300,6 +310,114 @@ func dataPlaneDelivery(name string, data delivery) *feature.Feature {
...
+			kafkacat.WithBootstrapServer(kafkaBootstrapUrlPlain),
+			kafkacat.WithTopic(topicName),
+			kafkacat.WithKey(message.Key),
+			kafkacat.WithHeaders(message.Headers),
+			kafkacat.WithPayload(message.Payload),
+		}
+		env := environment.FromContext(ctx)
+		matchers = AllOf(
+			func(have event.Event) error {
+				kafkaSourceClientSet, err := kafkaclientset.NewForConfig(kc.Config)
```
Comment (lionelvillard) [inline test/rekt/features/kafkasource/data_plane.go:392]: You should be able to use the injection API to get the client from the context. Look under pkg/client/injection/client. You should be able to do something like kafkaclient.Get(ctx)

File: test/rekt/kafka_source_test.go [hunk 2]
```diff
@@ -59,7 +60,9 @@ func TestKafkaSource(t *testing.T) {
 		t.Fatalf("could not copy secret(%s): %v", kafkaSASLSecret, err)
 	}
 
-	for _, f := range kafkasource.DataPlaneDelivery() {
+	testingFeatures := kafkasource.DataPlaneDelivery()
+	testingFeatures = append(testingFeatures, kafkasource.MtDataPlaneDelivery(kafkaClient))
```
Comment (lionelvillard) [inline test/rekt/kafka_source_test.go:64]: This feature is only applicable when the KafkaSource under test is the multi-tenant one. Check if `os.Getenv("TEST_MT_SOURCE")` is set and add the MT features.

**Comment by lionelvillard** [comment]
/approve
/lgtm

thanks!

**Pull request merged**